modules/gdprEnforcement.js

```diff
@@ -47,58 +47,79 @@ const analyticsBlocked = [];
 
 let addedDeviceAccessHook = false;
 
+// Helps in stubbing these functions in unit tests.
+export const internal = {
+  getGvlidForBidAdapter,
+  getGvlidForUserIdModule,
+  getGvlidForAnalyticsAdapter
+};
+
 /**
- * Returns gvlId for Bid Adapters. If a bidder does not have an associated gvlId, it returns 'undefined'.
- * @param  {string=} bidderCode - The 'code' property on the Bidder spec.
- * @retuns {number} gvlId
+ * Returns GVL ID for a Bid adapter / an USERID submodule / an Analytics adapter.
+ * If modules of different types have the same moduleCode: For example, 'appnexus' is the code for both Bid adapter and Analytics adapter,
+ * then, we assume that their GVL IDs are same. This function first checks if GVL ID is defined for a Bid adapter, if not found, tries to find User ID
+ * submodule's GVL ID, if not found, tries to find Analytics adapter's GVL ID. In this process, as soon as it finds a GVL ID, it returns it
+ * without going to the next check.
+ * @param {{string|Object}} - module
+ * @return {number} - GVL ID
  */
-function getGvlid(bidderCode) {
-  let gvlid;
+export function getGvlid(module) {
+  let gvlid = null;
+  if (module) {
+    // Check user defined GVL Mapping in pbjs.setConfig()
+    const gvlMapping = config.getConfig('gvlMapping');
+
+    // For USER ID Module, we pass the submodule object itself as the "module" parameter, this check is required to grab the module code
+    const moduleCode = typeof module === 'string' ? module : module.name;
+
+    // Return GVL ID from user defined gvlMapping
+    if (gvlMapping && gvlMapping[moduleCode]) {
+      gvlid = gvlMapping[moduleCode];
+      return gvlid;
+    }
+
+    gvlid = internal.getGvlidForBidAdapter(moduleCode) || internal.getGvlidForUserIdModule(module) || internal.getGvlidForAnalyticsAdapter(moduleCode);
+  }
+  return gvlid;
+}
+
+/**
+ * Returns GVL ID for a bid adapter. If the adapter does not have an associated GVL ID, it returns 'undefined'.
```

Comment:
if there's not a gvlid found, you're returning `null`, but this comment says it returns `undefined`. based on the tests, seems like the comments should be updated

Reply:
done